Day 15: “The Wonderful World of Web Feeds” / Maureen Holland (#206)
matuzo/HTMHell: hell/adventcalendar/2025/15/index.md

---
title: "The Wonderful World of Web Feeds"
author: "Maureen Holland"
author_bio: "Lead front-end web developer at silverorange.com. Passionate about accessibility, performance, and biking around Scotland."
date: 2025-12-15
author_links:
  - label: "Work"
    url: "https://blog.silverorange.com/"
    link_label: "blog.silverorange.com"
  - label: "Personal"
    url: "https://maureenholland.ca/magpie/"
    link_label: "maureenholland.ca/magpie"
  - label: "Mastodon"
    url: "https://hachyderm.io/@maureenholland"
    link_label: "@maureenholland"

> intro: "<p>Web feeds are incredible! And a bit confusing! This post offers a peek behind the curtain of what makes a web feed and delights in the fact RSS Autodiscovery means you don’t need to know anything about that to subscribe</p>"

Reviewer: add period

```suggestion
intro: "<p>Web feeds are incredible! And a bit confusing! This post offers a peek behind the curtain of what makes a web feed and delights in the fact RSS Autodiscovery means you don’t need to know anything about that to subscribe.</p>"
```

image: "advent25_15"
---

> Web feeds are incredible! And a bit confusing! Why are the feed links often called “RSS”? And why is this “RSS” feed in an <code>atom.xml</code> file… hang on, what is <code>feed.json</code> for?  What are they even feeding into anyway?

Reviewer: remove double space.

```suggestion
Web feeds are incredible! And a bit confusing! Why are the feed links often called “RSS”? And why is this “RSS” feed in an <code>atom.xml</code> file… hang on, what is <code>feed.json</code> for? What are they even feeding into anyway?
```

To start, web feeds are often referred to as “RSS” because RSS is the oldest format. [RSS](https://www.rssboard.org/rss-specification) stands for Really Simple Syndication. It is an XML-based specification for web content syndication ([including podcasts](https://help.apple.com/itc/podcasts_connect/#/itcb54353390)).

> In [broadcast syndication](https://web.archive.org/web/20091009143514/http://www.museum.tv/archives/etv/S/htmlS/syndication/syndication.htm), networks sell reruns of their original shows to other platforms. In web content syndication, feeds package web content into a format that can “rerun” on a feed reader application. Notably, web content creators are not selling their content to feed readers. Most feeds and readers are free. A subscriber is in the middle, choosing what content they want to follow and what reader they want to follow it on.

Reviewer: As a non-native speaker I had to look up “syndication”. Which might be fine but it could be nice to explain it inline. I really like the comparison with broadcast syndication and think that could be expanded upon for this example.

Something along the lines of:

```suggestion
Syndication is the sale or licensing of material for publication or broadcasting by others. This system generall has three parts: a seller, a buyer, and a consumer. In [broadcast syndication](https://web.archive.org/web/20091009143514/http://www.museum.tv/archives/etv/S/htmlS/syndication/syndication.htm), networks sell reruns of their original shows to other platforms, the buyers. The consumer then watches the rerun on the buyer’s channel. In web content syndication feeds are the sellers. They package web content into a format that can be “rerun” by the buyer using a feed reader application. Notably, web content creators are not selling their content to feed readers. Most feeds and readers are free. With web feeds you’re both the buyer and the consumer. You choose what content you want to follow and what reader you want to follow it on.
```
This is just a suggestions of course :). Feel free to ignore or rewrite in your own style.

			I rebuild my personal site every few years. This time, I decided I wanted to go as minimal as possible. It's been the most enjoyable iteration.
			</description>
		</item>
	</channel>
</rss>
```

This XML file includes general info about the feed (`channel`) and lists one `item` which contains a `title`, `link`, `description`, publication date (`pubDate`), and globally unique identifier (`guid`).

A feed reader, like [Feedbin](https://feedbin.com/home), [NetNewsWire](https://netnewswire.com/), or [NewsBlur](https://www.newsblur.com/), is able to parse that information and serve it in a human-readable format that will look something like this: 

<img src="simple-feed-in-reader.png" width="1140" height="372" loading="lazy" alt="Black text on white background. Title: A Vanilla Personal Site. Description: I rebuild my personal site every few years. This time, I decided I wanted to go as minimal as possible. It’s been the most enjoyable iteration. Gray text: maureenholland.ca, Apr 18 2023.">

It uses the unique identifier to determine if an item in the feed is new. It can do this for any number of feeds, constantly updating a reading list of your favourite web content.

> Importantly, feed readers have no proprietary control over your feed list. If you are dissatisfied with your reader, you can export your feeds to an [OPML](https://opml.org/spec2.opml) (Outline Processor Markup Language) file and import them to a new reader later.

Reviewer: Oh I didn’t know that was possible, neat!
